Source/Core/Core/PowerPC/ConditionRegister.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include "Common/CommonTypes.h"
+
+namespace PowerPC
+{
+enum CRBits
+{
+  CR_SO = 1,
+  CR_EQ = 2,
+  CR_GT = 4,
+  CR_LT = 8,
+
+  CR_SO_BIT = 0,
+  CR_EQ_BIT = 1,
+  CR_GT_BIT = 2,
+  CR_LT_BIT = 3,
+};
+
+// Optimized CR implementation. Instead of storing CR in its PowerPC format
+// (4 bit value, SO/EQ/LT/GT), we store instead a 64 bit value for each of
+// the 8 CR register parts. This 64 bit value follows this format:
+//   - SO iff. bit 61 is set
+//   - EQ iff. lower 32 bits == 0
+//   - GT iff. (s64)cr_val > 0
+//   - LT iff. bit 62 is set
+//
+// This has the interesting property that sign-extending the result of an
+// operation from 32 to 64 bits results in a 64 bit value that works as a
+// CR value. Checking each part of CR is also fast, as it is equivalent to
+// testing one bit or the low 32 bit part of a register. And CR can still
+// be manipulated bit by bit fairly easily.
+struct ConditionRegister
+{
+  // convert flags into 64-bit CR values with a lookup table
+  static const std::array<u64, 16> s_crTable;
+
+  u64 fields[8];
+
+  // Convert between PPC and internal representation of CR.
+  static inline u64 PPCToInternal(u8 value)
```

Comment:
Functions defined in a class/struct/union declaration don't need to be declared `inline`. They're already inline by default. Ditto for other relevant occurrences.